Makefile

```diff
@@ -208,19 +208,42 @@ set-manifest-image:
 ## --------------------------------------
 
 RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
+RELEASE_DIR := out
+
+$(RELEASE_DIR):
+	mkdir -p $(RELEASE_DIR)/
 
 .PHONY: release
-release:  ## Builds and push container images using the latest git tag for the commit.
+release: clean-release  ## Builds and push container images using the latest git tag for the commit.
 	@if [ -z "${RELEASE_TAG}" ]; then echo "RELEASE_TAG is not set"; exit 1; fi
 	# Push the release image to the staging bucket first.
 	REGISTRY=$(STAGING_REGISTRY) TAG=$(RELEASE_TAG) \
 		$(MAKE) docker-build-all docker-push-all
 	# Set the manifest image to the production bucket.
 	MANIFEST_IMG=$(PROD_REGISTRY)/$(IMAGE_NAME) MANIFEST_TAG=$(RELEASE_TAG) \
 		$(MAKE) set-manifest-image
-	# Generate release artifacts.
-	mkdir -p out/
-	kustomize build config/default > out/infrastructure-components.yaml
+	$(MAKE) release-manifests
+	$(MAKE) release-binaries
+
+.PHONY: release-manifests
+release-manifests: $(RELEASE_DIR) ## Builds the manifests to publish with a release
+	kustomize build config/default > $(RELEASE_DIR)/infrastructure-components.yaml
+
+release-binaries: ## Builds the binaries to publish with a release
+	RELEASE_BINARY=./cmd/clusterawsadm GOOS=linux GOARCH=amd64 $(MAKE) release-binary
+	RELEASE_BINARY=./cmd/clusterawsadm GOOS=darwin GOARCH=amd64 $(MAKE) release-binary
+
+release-binary: $(RELEASE_DIR)
```

Comment:
Same here, since it's not defining a specific output path that exists post-build, shouldn't this be a phony target?

Reply:
Let's not do phone - 1 sec

Reply:
This can be `$(RELEASE_DIR)/$(RELEASE_BINARY)-$(GOOS)-$(GOARCH)`

Reply:
See e.g. https://github.com/heptio/velero/blob/73488cdc0c649dc7ec9a692eab934c14ceb3a4b6/Makefile#L18, https://github.com/heptio/velero/blob/73488cdc0c649dc7ec9a692eab934c14ceb3a4b6/Makefile#L40, https://github.com/heptio/velero/blob/73488cdc0c649dc7ec9a692eab934c14ceb3a4b6/Makefile#L43-L45, https://github.com/heptio/velero/blob/73488cdc0c649dc7ec9a692eab934c14ceb3a4b6/Makefile#L97

Reply:
Should we trust that if we aren't specifying input source(s)?

Reply:
In general, I'd prefer to avoid phony targets if we can. Given that we only produce a single binary now, we can err on the side of specificity and have `$(RELEASE_DIR)/clusterawsadm-$(GOOS)-$(GOARCH)` as the target name - does that alleviate your concern?

Reply:
I more meant that we are relying on the removal of RELEASE_DIR in the `release` target to ensure that we are building a new binary, since we do not specify source files to watch for updates to build the target.

Reply:
Oh I see what you're saying now. I wish there were a good way in a Makefile to associate a binary with go source files. Ok let's do phony for the time being.